app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -2840,10 +2843,23 @@ class BrowserTabViewModel @Inject constructor(
 
     @SuppressLint("RequiresFeature") // it's already checked in isBlobDownloadWebViewFeatureEnabled
     private fun postMessageToConvertBlobToDataUri(url: String) {
-        for ((key, value) in replyProxyMap) {
-            if (sameOrigin(url.removePrefix("blob:"), key)) {
-                value.postMessage(url)
-                return
+        appCoroutineScope.launch {
```

Comment:
nit: why not launch with `dispatchers.io()` instead of launching then immediately switching?

Reply:
I don't know if it's always safe to do `replyProxy.postMessage(url)` on IO.

Reply:
Yeah, I didn't want to do post on `io`, wanted to preserve the same threading we have before as much as possible.
However I think the `.launch{}` is gonna run in the `Default` dispatcher. @anikiki, should I move it to dislpatchers.main()?